services/classesService.js

```diff
@@ -0,0 +1,264 @@
+const classesDao = require("../models/classesDao");
+const { error } = require("../middleware/error");
+const { appDataSource } = require("../models/datasource");
+const axios = require('axios');
+const dotenv = require('dotenv');
+dotenv.config();
+
+const getClassesList = async (topCategoryName, subCategoryName, sortBy, search) => {
+  const ordering = async (sortBy) => {
+    const sortOptions = {
+      salesDesc: "ORDER BY c.sales DESC, c.id ASC",
+      newest: "ORDER BY c.created_at DESC, c.id ASC",
+      default: "ORDER BY c.id ASC",
+    };
+    return sortOptions[sortBy] || sortOptions.default;
+  };
+  const orderingQuery = await ordering(sortBy);
+  const searchQuery = search ? `AND (c.title like '%${search}%' OR c.content like '%${search}%')` : "";
+  const topCategoryQuery = topCategoryName ? `AND t.name = '${topCategoryName}'` : "";
+  const subCategoryQuery = subCategoryName ? `AND s.name = '${subCategoryName}' AND t.name = '${topCategoryName}'` : "";
+  const limit = "";
+
+  const result = await classesDao.getClassesList(
+    topCategoryQuery,
+    subCategoryQuery,
+    searchQuery,
+    orderingQuery,
+    limit
+  );
+
+  let result2 = [];
+  if (topCategoryName) {
+    result2 = await classesDao.getSubCategories(topCategoryName);
+  };
+
+  return { classList: result, subCategoriesName: result2 };
+};
+const getUpcomingClasses = async() =>{
+  const salesDesc = "ORDER BY c.sales DESC, c.id ASC";
+  const newest = "ORDER BY c.created_at DESC, c.id ASC";
+  const limit = 'LIMIT 5'
+  
+  const salesOrder = await classesDao.getClassesList('', '', '', salesDesc,limit);
+  const newProductOrder = await classesDao.getClassesList('', '', '', newest,limit);
+  const upcomingClasses = await classesDao.getUpcomingClasses();
+  return { salesOrder, newProductOrder, upcomingClasses }
+};
+const getClassDetail = async(classId) => {
+  const [result] = await classesDao.getClassDetailByClassId(classId);
+  const scheduleInfoString = result.schedule_info;
+  const cleanedScheduleInfoString = scheduleInfoString.replace(/\//g, '');
+  let scheduleInfoArray = [];
+
+  if (cleanedScheduleInfoString) {
+    try {
+      scheduleInfoArray = JSON.parse(`[${cleanedScheduleInfoString}]`);
+    } catch (error) {
+      console.error('스케줄 정보 파싱 오류:', error);
+    }
+  }
+  const structuredData = {
+    id: result.id,
+    title: result.title,
+    name: result.name,
+    summary: result.summary,
+    content: result.content,
+    price: result.price,
+    top_category_name: result.top_category_name,
+    sub_category_name: result.sub_category_name,
+    hostId: result.hostId,
+    schedule_info: scheduleInfoArray,
+    main_image_source: result.main_image_source,
+    sub_image_source: result.sub_image_source,
+    address: result.address,
+    latitude: result.latitude,
+    longitude: result.longitude
+  };
```

Comment:
```suggestion
  const structuredData = { ...result, schedule_info: scheduleInfoArray };
```

spread 연산자 한 번 확인해 주세요